checker/src/org/checkerframework/checker/index/samelen/SameLenVisitor.java

```diff
@@ -38,6 +38,7 @@ protected void commonAssignmentCheck(
             /*@CompilerMessageKey*/ String errorKey) {
         if (valueType.getKind() == TypeKind.ARRAY
                 && TreeUtils.isExpressionTree(valueTree)
+                && !(valueTree.getKind() == Tree.Kind.NEW_ARRAY)
```

Comment:
On line 52 (below), call `isReceiverToStringParsable(rec)` instead of `!(rec instanceof Unknown)`.